cassandra-analytics-core/src/test/java/org/apache/cassandra/spark/validation/KeyStoreValidationTests.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.cassandra.spark.validation;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import org.apache.cassandra.secrets.SecretsProvider;
+import org.apache.cassandra.secrets.TestSecretsProvider;
+
+/**
+ * Unit tests that cover startup validation of a KeyStore
+ */
+public class KeyStoreValidationTests
+{
+    @Test()
+    public void testNullSecrets()
+    {
+        KeyStoreValidation validation = new KeyStoreValidation(null);
+
+        Assertions.assertThrows(RuntimeException.class, validation::perform);
```

Comment:
can you assert on the thrown exception's message, and for the other test cases? There could be other places that throw `RuntimeException`.

You can achieve such assertion with this code example

```java
        RuntimeException rte = Assertions.assertThrows(RuntimeException.class, validation::perform);
        Assertions.assertTrue(rte.getMessage().contains("blabla"));
```

Reply:
+1 , I think we want to assert the exception on the inner exception that RuntimeException is wrapping

Reply:
Added assertions on the message and the cause here and in 12 other places.